pkg/sentry/socket/plugin/config.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"golang.org/x/sys/unix"
+	"gvisor.dev/gvisor/pkg/seccomp"
+)
+
+// PluginFilters defines seccomp allowed rules that are needed by cgo.
```

Comment:
If these extra filters are needed due to difference in the cgo runtime, as opposed to pure Go, then they are not network-plugin-stack-related and should therefore be somewhere more generic (like `runsc/boot/filter/config/`).

It sounds like some of these are network-stack-related though (like the comment below that says `// alloc_seg in DPDK`), in which case these filters should be split into two files: one for the cgo-specific filters, and another file (which can be this current file) for the network-stack-related ones.

Reply:
Thanks for pointing out. Yes, I agree these two seccomp rule sets should be separated. We have now modified this part: CGO seccomp rules are defined in runsc/filter/config/config_cgo.go and runsc/fsgofer/filter/config_go.go.